app/src/main/java/com/android/solvit/provider/ui/NotificationsScreen.kt

```diff
@@ -30,32 +52,66 @@ fun NotificationScreen(
   LaunchedEffect(providerId) { viewModel.init(providerId) }
```

Comment:
If fetching notifications takes time, consider maybe adding a progress indicator so the user knows its loading rather than crashed maybe something like this : 
`val isLoading = remember { mutableStateOf(false) }

LaunchedEffect(providerId) {
    isLoading.value = true
    viewModel.init(providerId)
    isLoading.value = false
}

if (isLoading.value) {
    CircularProgressIndicator(modifier = Modifier.align(Alignment.Center))
} else {
    // Display notifications or empty state
}
`

Reply:
Good remark! I’ll definitely add a progress indicator to handle cases where fetching notifications takes time.Thanks for pointing that out!